### PR TITLE
Retrieve and build protobuf from remote URLs

### LINKED ISSUE #59: [enhancement] Retrieve and build protobuf from remote URLs
## Description

Currently, this repository is using cargo git dependencies to reference the proto packages from other ESDV projects. This has caused compatibility issues and needs to be updated. The dependent projects are using submodules to bring in their proto files, but this is also undesirable. The proposed solution is to retrieve the raw proto content with a URL and build from the `OUT_DIR`, similar to how UProtocol builds their protobuf files: https://github.com/eclipse-uprotocol/up-rust/blob/main/build.rs

See also: https://github.com/eclipse-ibeji/freyja/issues/157

## Acceptance criteria

- [ ] Proto files are retrieved from a remote URL and built from the `OUT_DIR`

### PR DESCRIPTION
**Pull request by wilyle**
Currently, this project is using cargo git dependencies to reference the proto packages from other ESDV projects such as Freyja. This has caused compatibility issues and needs to be updated. This PR makes a change to pull the content from a remote URL and build the proto files within the repo. Inspired by UProtocol: https://github.com/eclipse-uprotocol/up-rust/blob/main/build.rs

Fixes #59

### CODE REVIEW

File: proto/common/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+// This module contains constants related to remote protobuf files as well as helpers for referencing them.
+// This also helps manage interface versions from a central location, which is particularly helpful for Ibeji since
+// the interfaces are referenced in two different crates.
+
+pub const GITHUB_BASE_URL: &str = "https://raw.githubusercontent.com";
+
+pub mod freyja {
+    pub const REPO_NAME: &str = "eclipse-ibeji/freyja";
+    pub const VERSION: &str = "0.1.0";
```
Comment (ashbeitz) [inline proto/common/src/lib.rs:13]: REPO_VERSION

**Review by ashbeitz — COMMENTED** [review]
One minor comment.  Please take a look.

File: proto/common/src/lib.rs (hunk 1 above)
Comment (wilyle) [reply proto/common/src/lib.rs:13]: this is the same convention as the Freyja repo, and the `interface_url!` macro won't work if we rename this

Comment (ashbeitz) [reply proto/common/src/lib.rs:13]: Okay.  Leave as is then.

**Review by ashbeitz — APPROVED** [review]
Approved

**Pull request merged**